### PR TITLE
project1.txt

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -197,6 +197,8 @@ void thread_complete(){
 
     int ret_val = sigsetjmp(threads[current_thread_id].env,1);
     if (ret_val == 1) {
+	// Reset the quantum
+	setitimer(ITIMER_VIRTUAL, &timer, nullptr);
 	enable_interrupts();
```
Comment (ghost) [inline main.cpp:202]: Resets the quantum right before the thread returns to where ever it was pre-empted before

File: main.cpp [hunk 2]
```diff
@@ -490,6 +496,9 @@ int uthread_terminate(int tid) {
  * @param time_slice - the new time slice for each thread in microseconds
  */
 int uthread_init(int time_slice) {
+    if(time_slice <= 0) 
+	return -1;
```
Comment (ghost) [inline main.cpp:500]: Make sure time slice is valid

Comment (elliott-beach) [reply main.cpp:500]: nice

**Comment by elliott-beach** [comment]
Writeup looks good. Thanks.

**Comment by elliott-beach** [comment]
merged

**Pull request closed without merging**